lib/_stream_transform.js

```diff
@@ -217,3 +230,110 @@ function done(stream, er, data) {
     throw new ERR_TRANSFORM_ALREADY_TRANSFORMING();
   return stream.push(null);
 }
+
+function SourceIterator(asyncGeneratorFn, opts) {
+  const source = this;
+  this[kSourceIteratorPull] = null;
+  this[kSourceIteratorChunk] = null;
+  this[kSourceIteratorResolve] = null;
+  this[kSourceIteratorGenerator] = asyncGeneratorFn;
+  this[kSourceIteratorStream] = new Transform({
+    objectMode: true,
+    ...opts,
+    transform(chunk, encoding, cb) {
+      source.encoding = encoding ||
+        source[kSourceIteratorStream]._transformState.writeencoding;
```

Comment:
In what cases does this fallback apply? The `encoding` parameter is always present and non-empty, as far as I can tell.

Reply:
thanks for confirming, this was just defensive in case there was every a chance that it wasn't

Reply:
```suggestion
```